src/main/scala/io/shiftleft/fuzzyc2cpg/FuzzyC2Cpg.scala

```diff
@@ -113,6 +125,10 @@ class FuzzyC2Cpg(outputModuleFactory: CpgOutputModuleFactory) {
 
 }
 
+object FuzzyC2CpgCache {
+  val emptyFunctions = new mutable.HashMap[String, (String, CpgStruct.Builder)]()
```

Comment:
I think you need some kind of locking mechanism for this hash map.

Reply:
Yes, you're right, this thing needs to be locked, oops.

Reply:
I think it is not a good idea to do the locking like this. Especially because multiple persons are working on this frontend and like this it is super easy to forget the locking.
I suggest we make the `emptyFunctions` a private field, forward all the operations required and enforce the locking in the forwarding.

Reply:
There is absolutely _no_ other case in which this variable will be touched, and while, of course, you are right that we can make this cleaner and put even more hours into this small change, these kinds of reviews mostly block our progress. I'm not going to argue this though and just do as you say.